docs/use-cookie.md

```diff
@@ -0,0 +1,81 @@
+# useCookie
+
+Adds getters to easily read cookies and provides a simple API to set and clear cookies.
+
+Getters are then automatically defined by the mixin.
+- Getter name is a camelized conversion of the cookie name with the `Cookie` suffix. `user_id`  becomes `userIdCookie`
+
+**Options:**
+
+| Option| Description | Default value |
+|-----------------------|-------------|---------------------|
+| `expires` | Expire date of a cookie in days. Can be overriden on the options object on a per cookie basis |30|
+
+
+**Example**
+```js
+useCookie(this, { expires: 30 }) // Same as useCookie(this)
+```
+
+## Usage
+
+**Composing**
+
+```js
+import { Controller } from '@hotwired/stimulus'
+import { useCookie } from 'stimulus-use'
+
+export default class extends Controller {
+
+  connect() {
+    useCookie(this)
+    // Or pass an object to change the expires value (default 30 days):
+    // useCookie(this, { expires: 30 })
+
+    // Getters are initialized for all current cookies.
+    // For a cookie with name 'colorScheme' and value 'dark'
+    this.colorSchemeCookie // dark
```

Comment:
I'm not sure if this is the best behavior. I can technically see how this is useful, but this somehow goes against the other Stimulus/Stimulus use patterns.

I think it would make sense if you have to explicit tell the mixin the cookie names you want to use. This makes it less magical and easier to follow the flow, especially if other people on the team want to understand the controller. But also for us, so we can appropriately generate the right getters and setters for all the defined cookies.